### PR TITLE
New result wrapper

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: django_celery_extensions/config.py [hunk 1]
```diff
@@ -15,7 +15,6 @@
     'DEFAULT_TASK_STALE_TIME_LIMIT': None,
     'DEFAULT_TASK_MAX_QUEUE_WAITING_TIME': None,
     'AUTO_GENERATE_TASKS_DJANGO_COMMANDS': {},
-    'AUTO_GENERATE_TASKS_BASE': 'django_celery_extensions.task.DjangoTask',
```
Comment (matllubos) [inline django_celery_extensions/config.py:18]: unnecessary you can set default task parent in celery

File: django_celery_extensions/task.py [hunk 2]
```diff
@@ -67,91 +67,72 @@ def task_id(self):
         return self.id
 
 
-class OnCommitAsyncResult:
+class ResultWrapper:
```
Comment (matllubos) [inline django_celery_extensions/task.py:71]: every result has wrapper now. The wrapper keep informations about invocation. In the security library is whis result wrapper used for invocation log

File: django_celery_extensions/task.py [hunk 3]
```diff
@@ -168,6 +149,8 @@ class DjangoTask(Task):
 
     ignore_task_after_success_timedelta = None
 
+    result_wrapper_class = ResultWrapper
```
Comment (matllubos) [inline django_celery_extensions/task.py:153]: wrapper can be owerriden

File: django_celery_extensions/task.py [hunk 4]
```diff
@@ -324,6 +312,12 @@ def _get_ignore_task_after_success_key(self, task_args, task_kwargs):
             if self.ignore_task_after_success_timedelta else None
         )
 
+    def is_processing(self, args=None, kwargs=None):
```
Comment (matllubos) [inline django_celery_extensions/task.py:325]: is processing was moved from security. Now it is used unique key and cache to get information about running task. Only unique task can be evaluated now. It is much faster than searching in the log

File: django_celery_extensions/task.py [hunk 5]
```diff
@@ -571,7 +545,9 @@ def retry(self, args=None, kwargs=None, exc=None, throw=True,
         if not eta:
             eta = now() + timedelta(seconds=countdown)
 
-        self.on_task_retry(self.request.id, args, kwargs, exc, eta)
+        if max_retries is None or self.request.retries < max_retries:
+            # In the opposite way task will be failed
+            self.on_task_retry(self.request.id, args, kwargs, exc, eta)
```
Comment (matllubos) [inline django_celery_extensions/task.py:560]: there was bug, on_retry was called when max_retries was exceeded. In this situation only on_failed should be raised

**Review by rubickcz — APPROVED** [review]

**Review by Formulka — APPROVED** [review]

**Pull request merged**